### PR TITLE
ci: add `clippy` job to `rust.yml` workflow

### LINKED ISSUE #1950: Add CI check for `clippy` issues

### PR DESCRIPTION
**Pull request by hamirmahal**
closes #1950.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -2,14 +2,19 @@ name: Rustlings Tests
...
+    branches: [main]
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
Comment (hamirmahal) [inline .github/workflows/rust.yml:16]: I intentionally used `actions/checkout@v4` here because `actions/checkout@v3` uses a deprecated `Node.js` version.

Comment (hamirmahal) [reply .github/workflows/rust.yml:16]: I'm happy to take a look at that in a separate pull request.

**Comment by hamirmahal** [comment]
@mo8it what are your thoughts on this pull request?

**Comment by mo8it** [comment]
Hi @hamirmahal, I am a new maintainer and didn't deal with the CI configuration yet. My priority is to release version 6 soon. I will look at this PR afterwards.

**Pull request merged**